packages/core/review-workflows/server/src/services/document-service-middleware.ts

```diff
@@ -122,7 +123,34 @@ const handleStageOnUpdate: Middleware = async (ctx, next) => {
   return next();
 };
 
+/**
+ * Check if the entity is at the required stage before publish
+ */
+const checkStageBeforePublish: Middleware = async (ctx, next) => {
+  if (ctx.action !== 'publish') {
+    return next();
+  }
```

Comment:
Did you check if action === publish is enough to handle bulk publish and publishing via releases?

Reply:
Yes! It's enough for both cases so we are safe!